runtime/src/main/java/com/dylibso/chicory/runtime/MemCopyWorkaround.java

```diff
@@ -8,27 +8,33 @@ private interface MemoryCopyFunc {
         void apply(int destination, int offset, int size, Memory memory);
     }
 
-    static {
-        String workaround = System.getProperty("chicory.enableMemCopyWorkaround");
-
-        boolean enableMemCopyWorkaround;
-        if (workaround != null) {
-            enableMemCopyWorkaround = Boolean.parseBoolean(workaround);
+    public static boolean shouldUseMemWorkaround() {
+        String version = System.getProperty("java.version");
+        if (version.equals("0")) {
+            // Android https://developer.android.com/reference/java/lang/System#getProperties()
+            return false;
+        } else if (version.startsWith("1.")) {
+            // Java 8 or earlier: "1.8.0_231" → 8
+            return true;
         } else {
-            enableMemCopyWorkaround = Runtime.version().feature() < 21;
+            // Java 9 or later: "11.0.9" or "17.0.1"
+            int dotIndex = version.indexOf(".");
+            String majorStr = (dotIndex != -1) ? version.substring(0, dotIndex) : version;
+            return Integer.parseInt(majorStr) <= 17;
         }
+    }
 
-        if (enableMemCopyWorkaround) {
-            MemoryCopyFunc noop1 = (destination, offset, size, memory) -> {};
-            MemoryCopyFunc noop2 = (destination, offset, size, memory) -> {};
-            // Warm up the JIT... to make it see memoryCopyFunc.apply is megamorphic
-            for (int i = 0; i < 1000; i++) {
-                memoryCopyFunc = noop1;
-                MemCopyWorkaround.memoryCopy(0, 0, 0, null);
-                memoryCopyFunc = noop2;
-                MemCopyWorkaround.memoryCopy(0, 0, 0, null);
-            }
+    static {
+        MemoryCopyFunc noop1 = (destination, offset, size, memory) -> {};
+        MemoryCopyFunc noop2 = (destination, offset, size, memory) -> {};
+        // Warm up the JIT... to make it see memoryCopyFunc.apply is megamorphic
+        for (int i = 0; i < 1000; i++) {
```

Comment:
We should skip doing this loop if the workaround is not enabled.  No point in paying this class startup cost if it's not needed.